include/Options/Options.php

```diff
@@ -490,4 +499,42 @@ public function offsetSet( $offset, $value ): void {
 	public function offsetUnset( $offset ): void {
 		$this->reset( (string) $offset );
 	}
+
+	/**
+	 * Returns the list of modified sites.
+	 * On multisite, sites are cached.
+	 * /!\ At this point, some sites may have been deleted. They are removed from `$this->modified` here.
+	 *
+	 * @since 3.7
+	 *
+	 * @return bool[]
+	 * @phpstan-return array<int, true>
+	 */
+	private function get_modified(): array {
+		if ( empty( $this->modified ) ) {
+			// Not modified.
+			return $this->modified;
+		}
+
+		// Cleanup deleted sites and cache existing ones.
+		if ( ! is_multisite() || count( $this->modified ) === 1 ) {
+			// Not multisite, or multisite with only 1 site: no need to cache or verify existence.
+			return $this->modified;
+		}
+
+		// Fetch all the data instead of only the IDs, so it is cached.
+		$sites = get_sites(
+			array(
+				'site__in' => array_keys( $this->modified ),
+				'number'   => count( $this->modified ),
+			)
+		);
+
+		// Keep only existing blogs.
+		foreach ( $sites as $site ) {
+			$this->modified[ $site->id ] = true;
+		}
```

Comment:
It looks like that the code doesn't remove deleted sites as described in method comment:
```suggestion
		// Keep only existing blogs.
		$this->modifed = array();
		foreach ( $sites as $site ) {
			$this->modified[ $site->id ] = true;
		}
```

Reply:
Ah! Good catch. I think this line disappeared while I reworked this.

Reply:
Fixed in [79df102](https://github.com/polylang/polylang/pull/1498/commits/79df1025bdfe8599a5c7c177e36ae8aba660addc) and [7861f45](https://github.com/polylang/polylang/pull/1498/commits/7861f4534dfb0203d3d3abc1bf3e16c4fde6cc3e).